server/src/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -2435,4 +2460,66 @@ public boolean isSrcNatIpRequired(long vpcOfferingId) {
         final Map<Network.Service, Set<Network.Provider>> vpcOffSvcProvidersMap = getVpcOffSvcProvidersMap(vpcOfferingId);
         return vpcOffSvcProvidersMap.get(Network.Service.SourceNat).contains(Network.Provider.VPCVirtualRouter);
     }
+
+    private boolean rollingRestartVpc(final Vpc vpc, final List<DomainRouterVO> oldRouters, final ReservationContext context) throws ResourceUnavailableException, ConcurrentOperationException, InsufficientCapacityException {
```

Comment:
This method has a good chunk of code duplicated with `rollingRestartRouters`. Would mind extracting the code common code to methods and then reuse them?